src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1101,6 +1134,18 @@ private ExpressionEntry HandleCall(MethodDesc callee, MethodSignature signature,
             }
             var pointerSize = _compilation.NodeFactory.Target.PointerSize;
 
+            if (callee != null)
+            {
+                MetadataType calleeType = (MetadataType)callee.OwningType;
+                if (!calleeType.IsBeforeFieldInit)
```

Comment:
This should be triggered at the beginning of the method, not at the callsite. This is e.g. to make sure we're covered even if the first call happens through a delegate.

Reply:
Done. This probably improves size a bit too.